### ISSUE TITLE
this is awful

### ISSUE THREAD

**Issue by thcrt** [meithecatte/bashfuck#1]
you should never have done this. i hate it. good job

**Comment by meithecatte** [comment]
this is intended behavior, closing as wontfix :+1:

**Comment by 4censord** [comment]
i hate it, well done

**Comment by xplshn** [comment]
Will there be a version that works with DASH? (and Buysbox ash if possible :))?

**Comment by thcrt** [comment]
Who cares about DASH? I want Fish compatibility ;)

**Comment by dmgolembiowski** [comment]
Wondering which versions of BASH are bashfuck compatible? Asking for a friend.

**Comment by meithecatte** [comment]
I can tell you that it has been tested to work on 5.2.32. Apart from that, I can suggest that you try it and see! Report back with what you find ;3

**Comment by magicdrums** [comment]
Can't work with zsh... :-(

**Comment by Rouzan407** [comment]
can you give me me the open source version? because the generated output is still can deobfuscated

**Comment by meithecatte** [comment]
> can you give me me the open source version? because the generated output is still can deobfuscated

what do you mean? there is `encoder-payload.sh` in the repository already

**Comment by Rouzan407** [comment]
yes, but the generated output of bashfuck is still can be decompiled, do
you understand what mean? if i get your bashfuck code as open source i will
modify the code in order to cannot be decompiled

Pada Rab, 7 Mei 2025 08.52, Maja Kądziołka ***@***.***>
menulis:

**Comment by meithecatte** [comment]
The code in the repository is already open source. It is under an open source license, and it is in the form I created it.

The semicolons might have been newlines when I was working on it, and I may have had some notes as to what each count of underscores signifies, but that's minor stuff.

As for modifying the encoding scheme to make it not possible to decode, I don't see how that would ever be possible. One way or another, the code is going to have to be executed.
